doc/source/cluster/kubernetes/getting-started.ipynb

```diff
@@ -244,6 +258,164 @@
+    "Note that in production scenarios, you will want to use larger Ray pods. In fact, it is advantageous to size each Ray pod to take up an entire Kubernetes node. See the [configuration guide](kuberay-config) for more details."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "8f61dbc3",
+   "metadata": {},
+   "source": [
+    "#### Deploying a network policy for the static Ray cluster\n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "c2db0639",
+   "metadata": {},
+   "source": [
+    "If your Kubernetes has a default deny network policy for pods, you need to manually create a network policy to allow the bidirectional communication among the head and worker nodes in the Ray cluster as mentioned in [the Ports configurations doc](https://docs.ray.io/en/latest/ray-core/configure.html#ports-configurations)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "fcf57a8b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create a sample network policy for the static Ray cluster from the Ray repo:\n",
+    "! kubectl apply -f https://raw.githubusercontent.com/ray-project/ray/master/doc/source/cluster/kubernetes/configs/static-ray-cluster-networkpolicy.yaml"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "87686b72",
+   "metadata": {},
+   "source": [
+    "Once the network policy has been deployed, you can view the network policy for the static Ray cluster by running"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b5b8adcc",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! kubectl get networkpolicies\n",
+    "\n",
+    "# NAME                               POD-SELECTOR                           AGE\n",
+    "# ray-cluster-egress                 app=ray-cluster-head                   XXs\n",
+    "# ray-cluster-ingress                app=ray-cluster-head                   XXs"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "3178b34a",
+   "metadata": {},
+   "source": [
+    "#### External Redis Integration for fault tolerance\n",
+    "Ray by default uses an internal key-value store, called the Global Control Store (GCS). The GCS runs on the head node and stores cluster metadata. One drawback of this approach is that the head node loses the metadata if it crashes. \n",
+    "Ray can also write this metadata to an external Redis for reliability and high availability. With this setup, the static Ray cluster can recover from head node crasehs and tolerate GCS failures without losing connections to worker nodes.\n",
+    "\n",
+    "To use this feature, we need to pass in the `RAY_REDIS_ADDRESS` env var and `--redis-password` in the Ray head node section of [the Kubernetes deployment config file](https://raw.githubusercontent.com/ray-project/ray/master/doc/source/cluster/kubernetes/configs/static-ray-cluster.with-fault-tolerance.yaml)."
```

Comment:
you should also discuss the decision of using statically set port ranges for the worker ports. This is done because of the network policy. If there isn't the restriction, the worker port range is not required.

Reply:
Good point, let me add the discussion.

Reply:
Done, I added the context in [this commit](https://github.com/ray-project/ray/pull/31403/commits/8b9fcfc7dc78a420c0ef718fd1ff2067bad2c086).